src/sage/matrix/matrix_cmr_sparse.pyx

```diff
@@ -438,8 +438,106 @@ cdef class Matrix_cmr_chr_sparse(Matrix_cmr_sparse):
+        second_col1 = second_mat.columns()[second_col_index1]
+        second_col2 = second_mat.columns()[second_col_index2]
+        fir_nrows = range(fr)
+        sec_nrows = range(sr)
+        valid1 = False
+        valid2 = False
+        for i in fir_nrows:
+            if (first_col1[i] == 1 and first_col2[i] == 0) or (first_col1[i] == 0 and first_col2[i] == 1):
+                subcol1 = tuple(first_col1[k] for k in fir_nrows if k != i)
+                subcol2 = tuple(first_col2[k] for k in fir_nrows if k != i)
+                if subcol1 == subcol2:
+                    valid1 = True
+                    first_row_index = i 
+                    break
+        for i in sec_nrows:
+            if (second_col1[i] == 1 and second_col2[i] == 0) or (second_col1[i] == 0 and second_col2[i] == 1):
+                subcol1 = tuple(second_col1[k] for k in sec_nrows if k != i)
+                subcol2 = tuple(second_col2[k] for k in sec_nrows if k != i)
+                if subcol1 == subcol2:
+                    valid2 = True
+                    second_row_index = i 
+                    break
+        if not (valid1 and valid2):
+            raise ValueError('indicated columns of Matrices are not of appropriate form for 3-sum')
+        first_subcol = first_mat.delete_rows([first_row_index]).columns()[first_col_index1]
+        second_subcol = first_mat.delete_rows([second_row_index]).columns()[second_col_index1]
+        first_submat = first_mat.delete_columns([first_col_index1, first_col_index2])
+        second_submat = second_mat.delete_columns([second_col_index1, second_col_index2])
+        return first_submat
+        first_row = first_submat.rows()[first_row_index]
+        second_row = second_submat.rows()[second_row_index]
+        first_submat = first_submat.delete_rows([first_row_index])
+        second_submat = second_submat.delete_rows([second_row_index])
+        first_subrows = first_submat.rows()
+        second_subrows = second_submat.rows()
+        upper_right_rows = first_subcol.tensor_product(second_row).rows()
+        lower_left_rows = second_subcol.tensor_product(first_row).rows()
+        n1 = len(first_submat.rows())
+        n2 = len(second_submat.rows())
+        row_list = []
+        for i in range(n1):
+            l1 = []
+            r = first_subrows[i]
+            u = upper_right_rows[i]
+            for k in range(len(r)):
+                l1.append(r[k])
+            for k in range(len(u)):
+                l1.append(u[k])
+            row_list.append(l1)
+        for i in range(n2):
+            l2 = []
+            r = second_subrows[i]
+            u = lower_left_rows[i]
+            for k in range(len(r)):
+                l2.append(r[k])
+            for k in range(len(u)):
+                l2.append(u[k])
+            row_list.append(l2)
+        return Matrix_cmr_chr_sparse._from_data(upper_rows)     
+
+    def delete_rows(self, indices):
+        rows = self.rows()
```

Comment:
I think it would be better to implement a version of `matrix_from_rows()`, which is called by Sage's version of the `delete_rows()` method. See matrix1.pyx